linode/terraform-provider-linode
# new: Add `linode_nodebalancer_configs` data source

## 📝 Description

Add a new data source `linode_nodebalancer_configs` for user to query a list of configs under a nodebalancer. Although the list is not API filterable, we make some relevant fields client-filterable. 

Address issue #944.

## ✔️ How to Test

**Run integration test:**
`make testacc PKG_NAME="linode/nbconfigs"`

**Manual test:** 
1. In a sandbox environment (i.e. dx-devenv), run the following terraform script to create a nodebalancer resource and two different config resources under it
```
resource "linode_nodebalancer" "foobar" {
    label = "test_nb_configs"
    region = "us-east"
    client_conn_throttle = 20
    tags = ["tf_test"]
}

resource "linode_nodebalancer_config" "foo" {
    nodebalancer_id = "${linode_nodebalancer.foobar.id}"
    port = 8080
    protocol = "HttP"
    check = "http"
    check_passive = true
    check_path = "/"
}

resource "linode_nodebalancer_config" "bar" {
    nodebalancer_id = "${linode_nodebalancer.foobar.id}"
    # default port is 80
}
```
2. Create a `linode_nodebalancer_configs` data source to query the config that we want. Assume that we want the config which port is 80. Output the config's id.
```
data "linode_nodebalancer_configs" "specific_config" {
    nodebalancer_id = "${linode_nodebalancer.foobar.id}"
    filter {
        name = "port"
        values = ["80"]
    }
}

output "linode_nodebalancer_configs" {
  value = data.linode_nodebalancer_configs.specific_config.nodebalancer_configs.0.id  
}
```
3. Observe that the output id equals to the id of `linode_nodebalancer_config.bar`
